packages/rs-drive-abci/src/execution/types/state_transition_container/v0/mod.rs

```diff
@@ -1,58 +1,70 @@
 use dpp::consensus::ConsensusError;
 use dpp::state_transition::StateTransition;
 use dpp::ProtocolError;
+use std::time::Duration;
 
-/// This is a container that holds state transitions
+#[derive(Debug)]
+pub enum DecodedStateTransition<'a> {
+    SuccessfullyDecoded(SuccessfullyDecodedStateTransition<'a>),
+    InvalidEncoding(InvalidEncodedStateTransition<'a>),
+    FailedToDecode(FaultyStateTransition<'a>),
+}
 
-#[derive(Debug, Default)]
-pub struct StateTransitionContainerV0<'a> {
-    /// The asset lock state transitions
-    pub(super) valid_state_transitions: Vec<(&'a [u8], StateTransition)>,
-    /// Deserialization errors
-    pub(super) invalid_state_transitions: Vec<(&'a [u8], ConsensusError)>,
-    /// Deserialization errors that broke platform, these should not exist, but are still handled
-    pub(super) invalid_state_transitions_with_protocol_error: Vec<(&'a [u8], ProtocolError)>,
+#[derive(Debug)]
+pub struct InvalidEncodedStateTransition<'a> {
+    pub raw: &'a [u8],
+    pub error: ConsensusError,
+    pub elapsed_time: Duration,
+}
+
+#[derive(Debug)]
+pub struct FaultyStateTransition<'a> {
+    pub raw: &'a [u8],
+    pub error: ProtocolError,
+    pub elapsed_time: Duration,
+}
+
+#[derive(Debug)]
+pub struct SuccessfullyDecodedStateTransition<'a> {
+    pub decoded: StateTransition,
+    pub raw: &'a [u8],
+    pub elapsed_time: Duration,
 }
 
-pub trait StateTransitionContainerGettersV0<'a> {
-    fn valid_state_transitions(&'a self) -> &'a [(&'a [u8], StateTransition)];
-    fn invalid_state_transitions(&'a self) -> &'a [(&'a [u8], ConsensusError)];
-    fn invalid_state_transitions_with_protocol_error(&'a self) -> &'a [(&'a [u8], ProtocolError)];
+/// This is a container that holds state transitions
 
-    fn destructure(
-        self,
-    ) -> (
-        Vec<(&'a [u8], StateTransition)>,
-        Vec<(&'a [u8], ConsensusError)>,
-        Vec<(&'a [u8], ProtocolError)>,
-    );
+#[derive(Debug)]
+pub struct StateTransitionContainerV0<'a> {
+    state_transitions: Vec<DecodedStateTransition<'a>>,
```

Comment:
This is a refactoring, and I'm not sure why it is needed, because it should not be needed for the purpose of this PR.